chrome/content/zotero/elements/styleConfigurator.js

```diff
@@ -33,19 +31,26 @@
 				xmlns:xul="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"
 			>
 				<div>
-					<xul:richlistbox id="style-list" tabindex="0" />
+					<xul:richlistbox id="style-list" class="theme-listbox" tabindex="0" />
```

Comment:
This change added an unwanted focusring on Linux. Is it necessary?

![image](https://github.com/user-attachments/assets/4c2eda95-d860-4ace-a88f-ae23c3c754ff)

Reply:
The change is intentional, I followed our styling in the scaffold window. Since we are using it elsewhere, maybe we should consider adding rules for Linux.

Reply:
I see that you added `theme-listbox` to scaffold's `load.xhtml`. It's not used anywhere else, except for the multi-select `richlistbox` in scaffold, though not sure why. We have many richlistboxes in the codebase that do not use this, and generally for boxes that do not support multi-select there's no reason for the focusring to exist at all, and where `seltype=multi` is set, there is a css rule to add the focusring without adding the `theme-listbox` class.

All this to say that if there is no good actual styling reason for this class then it should be removed.

Reply:
It's initially from https://github.com/zotero/zotero/blob/c5027106c2cccb26e5b3092e3b87947fcce91e26/chrome/content/scaffold/scaffold.xhtml#L483 introduced by @AbeJellinek, I guess. These rules fix the looking of the listbox in darkmode on Windows after redesign.

I am not sure if applying these rules to all listbox elements would break anything. Any ideas @AbeJellinek ?

Reply:
The `theme-listbox` is from firefox: https://searchfox.org/mozilla-central/search?q=theme-listbox&path=&case=true&regexp=false , btw

Reply:
Yes, I know. XUL had a richlistbox and a listbox element, which I assume this is supposed to apply the style of listbox to the richlistbox element, which contains a superset of features of the old listbox. If there are issues with darkmode richlistbox on Windows, that should be addressed separately and directly, without breaking the look on other OSes.

Reply:
How about fixing the focus ring for linux? If the `theme-listbox` is incorrect in the doc prefs window, then it would be incorrect everywhere else.

We can address this separately for Windows, but that would be simply duplicating the `richlistbox.theme-listbox` rules for Windows.

Reply:
> How about fixing the focus ring for linux? If the theme-listbox is incorrect in the doc prefs window, then it would be incorrect everywhere else.

We do not use `theme-listbox` anywhere else, but scaffold for `richlistbox`. It would be incorrect everywhere else if used, because there should never be a focusring for `seltype=single` richlistbox items, and there's already a firefox css rule to add it for `seltype=multiple`. We don't have enough scaffold users on Linux for this to have been reported. Looking at scaffold now, it indeed looks wrong there.

> We can address this separately for Windows, but that would be simply duplicating the richlistbox.theme-listbox rules for Windows.

Are you sure it's not your own changes that fix specifically `theme-listbox` box for Windows, instead of any firefox rules (https://github.com/zotero/zotero/blob/b16bbc4a8ac40fa26b189cfe5fc76d6bd8109f51/scss/components/_richlistbox.scss#L1-L22)? Also, if `theme-listbox` is indeed required for richlistboxes to appear correctly on dark-mode on Windows, we would have had reports as they appear across the Zotero UI.

Reply:
Only `theme-listbox` uses these CSS variables. If without the class, we need to manually duplicate these rules all again.

I am thinking about removing all `theme-listbox` and maintaining a set of rules by ourselves since we expect all richlistbox elements to have the same looking.

Reply:
> Only `theme-listbox` uses these CSS variables. If without the class, we need to manually duplicate these rules all again.

Right, but what I am saying is that only `theme-listbox` richlistbox even looks wrong on darkmode windows, hence adding this class here wasn't necessary to make the richlistbox look good on darkmode windows, and also made it look bad on Linux.

Reply:
Using the `theme-listbox` has a historical reason. Tracing back to the time we tried to fix the styling on Windows after the redesign, the most important richlistitem is the one in the scaffold, which used `theme-listbox`, so the rules were designed for elements with this specific class. Later we found that other richlistitems were also not correct on Windows and borrowing the class name was the fastest solution.

More than once, I also thought about whether to remove the class from the scaffold and maintain unified rules for all richlistitem elements, during the redesign. At that time, rolling out a workable version had the highest priority. Now that we are heading to the target of being more stable and perfect, yes, I guess it's time to do this.

Reply:
> Tracing back to the time we tried to fix the styling on Windows after the redesign, the most important richlistitem is the one in the scaffold

Arguably the most important richlistbox which is seen by most users is precisely this one with the style picker.

Anyway, I just did some testing and it seems that indeed setting `theme-listbox` does make the richlistbox look better on Windows (though not correct without modifications in `_richlistbox.scss`). Still, the correct solution is to fix richlistbox appearance on Windows darkmode without using additional classes, and without breaking appearance of it on other platforms.